[DO NOT SQUASH MERGE] Fixes after community merge

roshan0708 opened the pull request:
### Description

Cherry-pick fixes for jdbc failures after latest community merge.

Engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/491

### Issues Resolved

Task: BABEL-5435

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

tanscorpio7 committed:
add tests for engine changes for BABEL-5335

Signed-off-by: Tanzeel Khan <tzlkhan@amazon.com>
(cherry picked from commit 28e5e7610f09cc008f69be21d60f63e9518765bc)

roshan0708 committed:
Fix test failure for bbf_role_admin_restrictions

Signed-off-by: Roshan Kanwar <rskanwar@amazon.com>

rishabhtanwar29 commented on test/JDBC/expected/bbf_role_admin_restrictions.out:
```diff
@@ -118,7 +118,7 @@ SET SESSION AUTHORIZATION bbf_role_admin;
 ~~ERROR (Code: 0)~~
 
-~~ERROR (Message: ERROR: permission denied to set session authorization
+~~ERROR (Message: ERROR: permission denied to set session authorization "bbf_role_admin"
```
So this change must be needed in gitfarm as well so let's keep this change separate from the rest of the changes by creating separate commits for each. The other change is being cherry-picked from gitfarm so it's anyways better to keep it separate.

roshan0708 replied:
@rishabhtanwar29 Right, for the test fix, it's already a separate commit in this PR, let's not squash merge this.

HarshLunagariya committed:
Update BABEL-1444 and Test-sp_reset_connection expected files (#3190)

After recent engine fixes, setting of role started giving expected output in case of parallel query enabled environment. This commit updates those test files expected output in case of parallel query enabled run.

Issues Resolved
Task: BABEL-5425

Authored-by: Harsh Lunagariya lunharsh@amazon.com
(cherry picked from commit 8b26542a71fbda6d9b0aa7e1008a2fca33fdfd98)
